internal/pkg/service/buffer/dependencies/dependencies.go

```diff
@@ -77,16 +77,13 @@ func NewServiceDeps(
 
 	// Create etcd client
 	etcdClient, err := etcdclient.New(
-		// Use separated context.
-		// Graceful shutdown is handled by Process.OnShutdown bellow.
-		// During the shutdown it is necessary to complete some etcd operations.
-		context.Background(),
+		ctx,
+		proc,
```

Comment:
`etcdclient` sa setupoval na zaciatku a nemal dorieseny graceful shutdown, ... resp. bolo to funkcne, ale inak ako mame v ostatnych castiach apky.

Teraz to riesi uz priamo `etcdclient.New`, dostane `proc` argument.

Okrem toho som vyhodil `WithConnectContext` parameter.

Teraz je to uz standardne:
- `ctx`, ktory ide do `etcdclient.New` je kontext pre vytvorenie spojenia
- a etcd client ma svoj `ctx`, .... a pri `shutdown` sa vola `client.Close`, co ukonci aj kontext klienta.